charts/consul/templates/server-config-configmap.yaml

```diff
@@ -56,7 +57,12 @@ data:
         "enabled": true
       },
       {{- end }}
-      "server": true
+      "server": true,
+      "leave_on_terminate": true,
```

Comment:
**suggestion:** These extra 'leave_on_terminate' and 'autopilot' settings should be removed as they were deemed destructive.

We need to check the other backports as anything from #3000 should not be in release/1.3.x, release/1.2.x and release/1.1.x (1.4.x is fine)

Reply:
Corrected as recommended by reverting back to `release/1.3.x` branch version of affected files.

```shell
$ git checkout 'release/1.3.x' -- charts/consul/templates/server-config-configmap.yaml
```
Re-applied `datadog-integration` changes into the following files:
* `charts/consul/templates/server-config-configmap.yaml`
  * Reincorporated `enable_debug` into `server.json` (updates `server-statefulset.yaml` config-checksum)
  * Reapplied all datadog and agent metric-related entries into the `telemetry-config.json` 
* `charts/consul/test/unit/server-statefulset.bats`
  * Updated `config-configmap` tests to reflect `enable_debug` update to `server.json` config
    * `"server/StatefulSet: adds config-checksum annotation when extraConfig is blank"`
    * `"server/StatefulSet: adds config-checksum annotation when extraConfig is provided"`
    * `"server/StatefulSet: adds config-checksum annotation when extraConfig is updated"`